src/procfile.rs

```diff
@@ -34,13 +34,13 @@ impl FromStr for Procfile {
 
     fn from_str(procfile_contents: &str) -> Result<Self, Self::Err> {
         let re_carriage_return_newline =
-            Regex::new("\\r\\n?").map_err(ProcfileParsingError::RegexError)?;
+            Regex::new("\\r\\n?").map_err(ProcfileParsingError::InvalidRegex)?;
         let re_multiple_newline =
-            Regex::new("\\n*\\z").map_err(ProcfileParsingError::RegexError)?;
+            Regex::new("\\n*\\z").map_err(ProcfileParsingError::InvalidRegex)?;
 
         // https://github.com/heroku/codon/blob/2613554383cb298076b4a722f4a1aa982ad757e6/lib/slug_compiler/slug.rb#L538-L545
         let re_procfile_entry = Regex::new("^[[:space:]]*([a-zA-Z0-9_-]+):?\\s+(.*)[[:space:]]*")
-            .map_err(ProcfileParsingError::RegexError)?;
+            .map_err(ProcfileParsingError::InvalidRegex)?;
```

Comment:
I would even go so far and say these arent `ProcfileParsingError`s. When they occur, it's clearly a programmer error on `procfile-cnb`s part. Those don't have to be errors at runtime IMO. This is one of the rare occurrences where I think panicing is fine.

I suggest we remove the `InvalidRegex` variant and call `expect` instead of mapping. Clippy will actually verify that the regex in our source is valid/compilable - so actual panics should never happen if our CI setup is correct. https://rust-lang.github.io/rust-clippy/master/#invalid_regex

This would make things much easier and would've prevented the incorrect error message in the first place.

Reply:
I also found https://github.com/Canop/lazy-regex which does compile-time regex validation - however I'm slightly reluctant to switch to a less popular crate that adds another layer of abstraction around `regex`.

Agree about Clippy validation, and I'd be fine with `.unwrap()` with a comment too.

Reply:
So once those errors are removed, `impl FromStr for Procfile` never returns an error.
As such, I was thinking I could just use `type Err = Infallible` (`std::convert::Infallible`).

However it seems conversion from infallible (without using eg `unwrap()`) is still pretty rough around the edges, eg this unstabilised feature:
https://github.com/rust-lang/rust/issues/61695

One option would be to drop the `FromStr` and just use the infallible `impl From<&str> for Procfile` instead?
Thoughts?

Reply:
From<&str> sounds good to me - if it's really infallible there is no need to encode errors.

Reply:
So thinking about this some more, rather than switch from `FromStr` to `impl From<&str> for Procfile` now, only to then have to switch to either `TryFrom` or back to `FromStr` in the future (when we fix #73), I'll use an empty error variant (`pub enum ProcfileParsingError {}`) instead.